pages/booking.js

```diff
@@ -0,0 +1,94 @@
+import { useState, useEffect } from "react";
+// import { useRouter } from "next/router";
+
+import BookingForm from "../components/BookingForm";
+import NumOfTickets from "../components/NumOfTickets";
+import PaymentModule from "../components/PaymentModule";
+import Seating from "../components/Seating";
+import StickyBooking from "../components/StickyBooking";
+import styles from "../styles/Booking.module.css";
+import BackBtn from "../components/BackBtn";
+
+export const getServerSideProps = async (context) => {
+  const value = Object.values(context.query);
+  const key = Object.keys(context.query);
+  const URL = `http://localhost:3000/api/bookings/${value}`;
+
+  if (key == "screeningId") {
```

Comment:
This comparison is a little bit risky. You are comparing `key` (set on line 14 to the return value of `Object.keys()` which is an array) to the string `"screeningId"`. Comparing an array and a string will sometimes be true, if the array contains exactly one item that equals the string.

This is one of those javascript peculiarities. In most other languages (including typescript) the comparison would be invalid, which makes sense. Comparing an array to a string is usually a mistake.

Instead, you could just do this:

```js
const value = context.query.screeningId;
if (value) {
  // ...
```

That just checks to make sure that the query parameter exists (i.e. is part of the URL), and then uses it's value from there on.